src/protocol/node.rs

```diff
@@ -0,0 +1,248 @@
+use std::io::{Read, Write};
+
+use anyhow::{bail, Context};
+use byteorder::{ReadBytesExt, WriteBytesExt};
+
+use crate::ident::Ident;
+use crate::protocol::{BoundedString, Decode, Encode, VarInt};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Node {
```

Comment:
Not sure if I like the name `Node`. Seems a bit vague. Might be better to call this `CommandNode` for clarity when used outside the module. Especially since this is only used in one place AFAICT.

Reply:
True, I just straight copied all names from https://wiki.vg/Protocol#Commands and https://wiki.vg/Command_Data for the moment.